### PR TITLE
feat: seaport royalties

### PR DESCRIPTION
**Pull request by lljxx1**

### CODE REVIEW

File: src/sync/events/storage/fill-events/index.ts [hunk 1]
```diff
@@ -24,6 +25,12 @@ export type Event = {
   usdPrice?: string;
   isPrimary?: boolean;
   baseEventParams: BaseEventParams;
+
+  royalty_fee_bps?: number;
```
Comment (georgeroman) [inline src/sync/events/storage/fill-events/index.ts:29]: We should use camelCase for the `Event` type.

**Review by georgeroman — CHANGES_REQUESTED** [review]

File: src/sync/events/handlers/royalties/index.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { Royalty } from "@/utils/royalties";
+import * as es from "@/events-sync/storage";
+
+import { logger } from "@/common/logger";
+import * as seaport from "@/events-sync/handlers/royalties/seaport";
+
+const registry = new Map<string, RoyaltyAdapter>();
+
+export type RoyaltyResult = {
+  royalty_fee_bps: number;
```
Comment (georgeroman) [inline src/sync/events/handlers/royalties/index.ts:10]: Let's use camelCase.

File: src/sync/events/handlers/royalties/seaport.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  for (let index = 0; index < eventsInfos.length; index++) {
+    const eventsInfo = eventsInfos[index];
+    const onchainData = await parseEventsInfo(eventsInfo);
+    allOnChainData.push(onchainData);
+  }
+  return allOnChainData;
+}
+
+export async function extractRoyalties(fillEvent: es.fills.Event) {
+  const royalty_fee_breakdown: Royalty[] = [];
```
Comment (georgeroman) [inline src/sync/events/handlers/royalties/seaport.ts:27]: Let's use camelCase.

File: src/sync/events/handlers/royalties/seaport.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        totalTransfers++;
+      }
+    }
+  }
+
+  for (const address in state) {
+    const { tokenBalanceState } = state[address];
+    const weth = "erc20:0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2";
+    const native = "native:0x0000000000000000000000000000000000000000";
+    const balanceChange = tokenBalanceState[native] || tokenBalanceState[weth];
```
Comment (georgeroman) [inline src/sync/events/handlers/royalties/seaport.ts:100]: Why only weth? Seaport does support any erc20.

File: src/sync/events/handlers/royalties/seaport.ts [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+  const result = {
+    txHash,
+    sale: {
+      tokenId,
+      contract,
+      price: formatEther(price),
+    },
+    totalTransfers,
+    royalty_fee_bps: getTotalRoyaltyBps(royalty_fee_breakdown),
```
Comment (georgeroman) [inline src/sync/events/handlers/royalties/seaport.ts:143]: Let's use camelCase.

**Review by georgeroman — CHANGES_REQUESTED** [review]

**Review by georgeroman — CHANGES_REQUESTED** [review]

File: src/sync/events/handlers/royalties/seaport.ts (hunk 4 above)
Comment (lljxx1) [reply src/sync/events/handlers/royalties/seaport.ts]: Sorry, I forget to push this changes

**Pull request merged**